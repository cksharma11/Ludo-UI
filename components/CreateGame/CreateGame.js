import React, { Fragment } from 'react';
import CreateGameStyles from './CreateGame.style';
import Header from '../Header/Header';
import labels from '../../config/labels/labels';

const CreateGame = () => {
  return (
    <Fragment>
      <Header title="Ludo" />
      <form method="POST" action="/createGame">
        <div className="activity-header">{labels.HOST_GAME}</div>
        <input type="text" name="hostName" placeholder="hostname" />
        <input
          type="number"
          name="numberOfPlayers"
          placeholder="number of players"
        />
        <input type="submit" value={labels.CREATE_GAME} />
      </form>
      <style jsx>{CreateGameStyles}</style>
    </Fragment>
  );
};

export default CreateGame;
