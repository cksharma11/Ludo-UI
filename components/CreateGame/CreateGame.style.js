import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const CreateGameStyles = css`
  form {
    font-family: fantasy;
    display: flex;
    height: 550px;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: ${colors.gray.shade2};
  }

  form > input {
    background: ${colors.gray.shade3};
    font-family: fantasy;
    width: 200px;
    height: 30px;
    border: 1px solid ${colors.gray.shade1};
    color: ${colors.orange.shade1};
    text-align: center;
    line-height: 2;
    text-decoration: none;
    padding: 5px;
    margin: 5px;
    border-radius: 2px;
  }

  .activity-header {
    font-family: fantasy;
    font-size: 30px;
  }
`;

export default CreateGameStyles;
