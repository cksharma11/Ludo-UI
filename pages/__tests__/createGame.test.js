import React from 'react';
import { shallow } from 'enzyme';
import CreateGame from '../createGame';

describe('createGame', () => {
  it('should render properly', () => {
    const wrapper = shallow(<CreateGame />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
