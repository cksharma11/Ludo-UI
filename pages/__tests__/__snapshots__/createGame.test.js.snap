// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createGame should render properly 1`] = `"<CreateGame />"`;
